packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -639,11 +642,47 @@ export function makeVirtualObjectManager(
   }
 
   function defineKind(tag, init, actualize, finish) {
-    return defineKindInternal(tag, init, actualize, finish, false);
+    const kindID = `${allocateExportID()}`;
+    return defineKindInternal(kindID, tag, init, actualize, finish, false);
+  }
+
+  let kindIDID;
+  const kindDescriptors = new WeakMap();
+
+  function reanimateDurableKindID(vobjID, _proforma) {
+    const { subid: kindID } = parseVatSlot(vobjID);
+    const raw = syscall.vatstoreGet(`vom.kind.${kindID}`);
+    assert(raw, X`unknown kind ID ${kindID}`);
+    const durableKindDescriptor = JSON.parse(raw);
+    const kindToken = Far('kind', {});
+    kindDescriptors.set(kindToken, durableKindDescriptor);
+    return kindToken;
+  }
+
+  function makeKindHandle(tag) {
+    if (!kindIDID) {
+      kindIDID = `${allocateExportID()}`;
+      syscall.vatstoreSet('kindIDID', kindIDID);
+      vrm.registerKind(kindIDID, reanimateDurableKindID, () => null, true);
+    }
+    const kindID = `${allocateExportID()}`;
+    const kindIDvref = `o+${kindIDID}/${kindID}`;
+    const durableKindDescriptor = { kindID, tag };
+    const kindHandle = Far('kind', {});
+    kindDescriptors.set(kindHandle, durableKindDescriptor);
+    registerValue(kindIDvref, kindHandle, false);
+    syscall.vatstoreSet(
+      `vom.kind.${kindID}`,
+      JSON.stringify(durableKindDescriptor),
+    );
+    return kindHandle;
   }
 
-  function defineDurableKind(tag, init, actualize, finish) {
-    return defineKindInternal(tag, init, actualize, finish, true);
+  function defineDurableKind(kindToken, init, actualize, finish) {
```

Comment:
```suggestion
  function defineDurableKind(kindHandle, init, actualize, finish) {
```
Is this right? If not, then I don't yet understand this code.

Reply:
Dang, I thought I'd caught all those but apparently a few sneaked through the pre-checkin copy edit pass.